WordPress/Classes/ViewRelated/Blog/TimezoneSelectorViewModel.swift

```diff
@@ -0,0 +1,203 @@
+        }
+    }
+
+    /// Helper method to fetch all TimezoneInfo objects from DB
+    func loadDataFromDB() -> [TimezoneInfo] {
+        let fetchRequest = NSFetchRequest<TimezoneInfo>(entityName: "TimezoneInfo")
+        fetchRequest.sortDescriptors = [NSSortDescriptor(key: "continent", ascending: true),
+                                        NSSortDescriptor(key: "label", ascending: true)]
+        do {
+            return try ContextManager.sharedInstance().mainContext.fetch(fetchRequest)
+        } catch {
+            print(error)
+            return []
+        }
+    }
+
+    /// Returns a sorted list of continents and timezones mapped to a continent
+    private func setupVariables(with result: [TimezoneInfo]) -> ContinentsAndTimezones {
+        var continentNames = result.reduce(into: Set<String>(), { (mySet, timezone) in
+            mySet.insert(timezone.continent)
+        }).sorted()
+
+        // remove manual offset's from it's sorted position and push it to the bottom
+        if let manualOffsetIndex = continentNames.index(of: self.MANUAL_OFFSET) {
+            continentNames.remove(at: manualOffsetIndex)
+            continentNames.append(self.MANUAL_OFFSET)
+        }
+        var timezoneNamesSortedByContinent: [String: [(String, String)]] = [:]
+        for continent in continentNames {
+            let allTimezonesInContinent = result.filter({ $0.continent == continent })
+
+            if continent == self.MANUAL_OFFSET {
+                // sort the UTC strings
+                let allLabelsAndValues = allTimezonesInContinent.map({ (label: $0.label, value: $0.value) })
+                timezoneNamesSortedByContinent[continent] = allLabelsAndValues.sorted(by: { (left, right) -> Bool in
+                    guard let leftNumString = left.value.components(separatedBy: "UTC").last,
+                        let rightNumString = right.value.components(separatedBy: "UTC").last,
+                        let floatLeftNum = Float(leftNumString),
+                        let floatRightNum = Float(rightNumString) else {
+                            return false
+                    }
+                    return floatLeftNum > floatRightNum
+                })
+            } else {
+                timezoneNamesSortedByContinent[continent] = allTimezonesInContinent.sorted(by: { (timezone1, timezone2) -> Bool in
+                    return timezone1.label < timezone2.label
+                }).map({ ($0.label, $0.value) })
+            }
+        }
+        return (headerNames: continentNames, sortedNamesByContinent: timezoneNamesSortedByContinent)
+    }
+
+    /// returns an IndexPath, which is the user's current selection
+    private func getIndexPathToHighlight(timezoneString: String?, manualOffset: NSNumber?, data: ContinentsAndTimezones, allTimezones: [TimezoneInfo]) -> IndexPath? {
+        if let usersTimeZone = timezoneString, !usersTimeZone.isEmpty {
+            for (continentIndex, continent) in data.0.enumerated() {
```

Comment:
I feel like all this code (for both timezones and manual offsets) could be simplified if we had a simple data type to represent these (which could implement `Equatable`) instead of using strings or numbers (which need to be transformed in order to compare them). Even if we don't do that, the manual offset comparison code could be extracted to a helper, which could also be used by the site settings screen when creating the timezone cell's label text.